src/plugins/platforms/freerdp/qfreerdppeer.cpp

```diff
@@ -979,7 +1032,18 @@ bool QFreeRdpPeer::init() {
 		return false;
 	}
 
-	peerCtx->event = new QSocketNotifier(mClient->sockfd, QSocketNotifier::Read);
+	/* peer->sockfd is -1 since freerdp 3ed4d09adcbdc1a7c7f537527265d4c479e7525c */
```

Comment:
The case with `peer->sockfd == -1` has already been treated in a previous commit (need rebasing ?) . The fix saves the original file descriptor.
The problem that could happen with `GetEventFileDescriptor` is that if at some point in FreeRDP we decide to add a check that the `HANDLE` is really an `Event` (and in in this case it's not), then we'll have a problem. We should probably have a generic  `GetHandleFileDescriptor` in WinPR (rather easy as everything is there for that)